sdk/aqueduct/artifacts/base_artifact.py

```diff
@@ -14,6 +14,7 @@ class BaseArtifact(ABC):
     _dag: DAG
     _content: Any
     _from_flow_run: bool
+    _from_local_data: bool = False
```

Comment:
I would rather have this information on the underlying DAG's ArtifactMetadata object. See how this was done in https://github.com/aqueducthq/aqueduct/pull/1093/files, specifically `sdk/aqueduct/models/artifact.py`. 

The reason being that it's easier to think about all the construction fields are on the same data plane. `_from_flow_run` is different since the underlying DAG is fetched from the server. `_from_operator_type` is in the wrong place imo, so let's not follow that example.